### PR TITLE
Add new navigation pattern for Value Tracking

### LINKED ISSUE #53821: Value Tracking should have a toggle button to disable single click navigation
Verbatim feedback from @davidwengier 

> Can there be a toolbar button to disable the single-click navigate? I feel like its jarring if I’m just exploring the tree

We could solve this by adding a button or possibly allowing hover context for information available on the right pane (file and line number)

### PR DESCRIPTION
**Pull request by ryzngard**
Value tracking originally used the standard WPF navigation pattern. This would also navigate to the source line based on selection.

This improves the design by manually handling keyboard navigation for most cases. By default, if an item is selected it no longer navigates to source. There are conditions where this is different:

| Shortcut | Expand Behavior | Navigates To Source | Previous or Next |
| --------- | ---------------- | ---------------------- | --------------------- |
| Arrow Down | False | True| Next |
| Arrow Up | False | True| Next |
| Arrow Left | Collapse Only | False | Neither |
| Arrow Right | Expand Only | False | Neither |
| Space | Toggle | False | Neither |
| Enter | False | True | Neither |
| F8 | Expand Only | True | Next |
| Shift + F8 | False | True | Previous |
| Click | False | True| Neither |
| Double Click | False | True | Neither |

Resolves #53821

### CODE REVIEW

File: src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs [hunk 1]
```diff
@@ -33,6 +40,97 @@ public bool IsLoading
...
+                    {
+                        return;
+                    }
+
+                    foreach (var item in items.Cast<TreeViewItemBase>())
+                    {
+                        item.Parent = parent;
+                    }
+                }
+            };
```
Comment (davidwengier) [inline src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs:68]: Call me old, but a 20 line lambda with a local function should probably just be a method 😛

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs [hunk 2]
```diff
@@ -32,9 +32,97 @@ public ValueTrackingTree(ValueTrackingTreeViewModel viewModel)
...
+
+            static bool TryToggleExpanded(TreeViewItemBase? node)
+                => TrySetExpanded(node, node is null ? false : !node.IsNodeExpanded);
+        }
+
+        private void ValueTrackingTreeView_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+        {
+            if (sender is TreeViewItemBase viewModel)
+            {
+                SelectItem(viewModel, true);
```
Comment (davidwengier) [inline src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:81]: Shouldn't this also expand, based on your table?

**Review by davidwengier — APPROVED** [review]

File: src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs (hunk 1 above)
Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs:68]: maybe... you don't always know when you start writing it :P

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs (hunk 2 above)
Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:81]: Actually, I think the table is wrong here. Imo clicking on an item shouldn't expand at all. Open to feedback on that behavior though

Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:81]: Updated table to reflect this behavior as written

File: src/VisualStudio/Core/Def/ValueTracking/TreeItemViewModel.cs [hunk 3]
```diff
@@ -70,6 +70,7 @@ internal class TreeItemViewModel : TreeViewItemBase
             Workspace = workspace;
             DocumentId = documentId;
 
+
```
[the remark this answers is not in the document]
Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/TreeItemViewModel.cs:73]: every... single... time...

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs (hunk 2 above)
Comment (davidwengier) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:81]: As the person who made double click on a project file open the editor, and not expand/collapse, I get the logic of it and support you, but you should probably expect feedback too :)

Comment (Cosifne) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:81]: I believe double-clicking in Test Explorer would do an expansion. And I just know solution explorer does a different thing😮

**Review by Cosifne — APPROVED** [review]

**Comment by mikadumont** [comment]
I do think arrow up, arrow down, and single click should navigate to source. We should keep behavior consistent with other results window like Find All References and shouldn't add keyboard friction causing the experience to be less seamless for users.

**Comment by ryzngard** [comment]
I've changed the table to match Mika's request and updated the code. If we get more user feedback on wanting this to be different we will reevaluate.

**Pull request merged**